### PR TITLE
ImagePuller refactoring

### PR DESCRIPTION
**Pull request by ronnielai**
A plain refactoring
- Moving image pullers to a new pkg/kubelet/images directory
- Hiding image pullers inside the new ImageManager 

The next step is to consolidate the logic of the serialized and the parallel image pullers inside ImageManager

xref: #25577

### CODE REVIEW

File: pkg/kubelet/images/impl.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ 		recorder: 		recorder,
+ 		runtime:  		runtime,
+ 		backOff:  		backOff,
+ 		imagePuller:   imagePuller,
+ 	}
+}
+
+func (*) EnsureImageExists(pod *api.Pod, container *api.Container, pullSecrets []api.Secret) (error, string) {
+	return imagePuller.pullImage(pod , container pullSecrets)
+}
```
Comment (vishh) [inline pkg/kubelet/images/impl.go]: nit: add a new line

File: pkg/kubelet/images/parallel_image_puller.go [hunk 2]
```diff
@@ -96,25 +97,25 @@ func (puller *imagePuller) PullImage(pod *api.Pod, container *api.Container, pul
...
 	if err := puller.runtime.PullImage(spec, pullSecrets); err != nil {
 		puller.logIt(ref, api.EventTypeWarning, events.FailedToPullImage, logPrefix, fmt.Sprintf("Failed to pull image %q: %v", container.Image, err), glog.Warning)
 		puller.backOff.Next(backOffKey, puller.backOff.Clock.Now())
-		if err == RegistryUnavailable {
+		if err == kubecontainer.RegistryUnavailable {
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return err, msg
 		} else {
-			return ErrImagePull, err.Error()
+			return kubecontainer.ErrImagePull, err.Error()
```
Comment (Random-Liu) [inline pkg/kubelet/images/parallel_image_puller.go]: In the future, we may also want to move the image pulling error to image manager?

Comment (vishh) [inline pkg/kubelet/images/parallel_image_puller.go]: +1. #25577 for reference

**Comment by vishh** [comment]
LGTM

**Comment by vishh** [comment]
@Random-Liu I applied the LGTM label. If you'd like to further review this PR, feel free to remove that label.

**Comment by Random-Liu** [comment]
LGTM.

File: pkg/kubelet/images/parallel_image_puller.go (hunk 2 above)
Comment (ronnielai) [inline pkg/kubelet/images/parallel_image_puller.go]: Tracked in #29122

**Comment by ronnielai** [comment]
@k8s-bot test this #IGNORE

**Comment by ronnielai** [comment]
Rebased. Reapplied LGTM.

**Pull request merged**